app/services/estate_visit_query.rb

```diff
@@ -14,18 +14,35 @@ def visits_to_print_by_slot(date)
     grouped_visits(visits)
   end
 
-  def processed(limit:, prisoner_number:)
+  def processed(limit:, query:)
     visits = Visit.preload(:prisoner, :visitors).
              processed.
              from_estates(@estates).
              order('visits.updated_at desc').limit(limit)
 
-    if prisoner_number.present?
-      number = Prisoner.normalise_number(prisoner_number)
-      visits = visits.joins(:prisoner).where(prisoners: { number: number })
-    end
+    return visits.to_a unless query
+    by_prisoner_number(query, visits) + by_human_id(query, visits)
+  end
+
+  def requested(query: nil)
+    visits = Visit.preload(:prisoner, :visitors).
+             with_processing_state(:requested).
+             from_estates(@estates).
+             order('created_at asc')
+
+    return visits.to_a unless query
+    by_prisoner_number(query, visits) + by_human_id(query, visits)
```

Comment:
I'd combine the 2 queries using a SQL `OR`, that way we'll avoid making an extra db call.